src/ui.js

```diff
@@ -302,6 +302,77 @@ const ui = {
     //console.log("un_hookup_ui", elem.id)
   },
 
+  scaleNode: (node, width, height = width) => {
+    node.setAttribute('x', 0)
+    node.setAttribute('y', 0)
+    const originalWidth = node.getAttribute('width')
+    const originalHeight = node.getAttribute('height')
+
+    const svgWrapper = document.createElementNS(SVG.ns, 'svg')
+    svgWrapper.setAttribute('width', width)
+    svgWrapper.setAttribute('height', height)
+    svgWrapper.setAttribute('viewBox', `0 0 ${originalWidth} ${originalHeight}`)
+    svgWrapper.appendChild(node)
+
+    return svgWrapper
+  },
+
+  getTimestamp: () => {
+    const now = new Date()
+    let hours = now.getHours()
+    let minutes = now.getMinutes()
+    let seconds = now.getSeconds()
+
+    if (minutes < 10) minutes = '0' + minutes
+    if (seconds < 10) seconds = '0' + seconds
+
+    return `${hours}:${minutes}:${seconds}`
+  },
+
+  addActivityLogItem: (verb, beforeNode, afterNode) => {
+    let li = document.createElement('li')
```

Comment:
this is a lot of html generation in JavaScript.  I think it can be shortened and made simpler by using the HTML <template> tag in the document then just cloning-and-pasting it in the <ol>

https://www.html5rocks.com/en/tutorials/webcomponents/template/